lib/idle_session_expiration_store.rb

```diff
@@ -0,0 +1,15 @@
+require 'redis-session-store'
+
+class IdleSessionExpirationStore < RedisSessionStore
+  def get_session(env, sid)
+    super.tap { update_session_ttl(sid) }
+  end
+
+  def update_session_ttl(sid)
+    expiry = default_options[:expire_after]
+    redis.expire(prefixed(sid), expiry)
+  rescue Errno::ECONNREFUSED, Redis::CannotConnectError => error
+    on_redis_down.call(error, env, sid) if on_redis_down
+    return false
+  end
+end
```

Comment:
I also tested that the expiry time automatically resets every time a new request is made. So, if a user signs in at 12:00:00, makes their last request 30 seconds later. The session times out at 12:01:30, not 12:01:00.